lib/Sema/CSDiag.cpp

```diff
@@ -5421,8 +5431,137 @@ bool FailureDiagnosis::visitApplyExpr(ApplyExpr *callExpr) {
     // Type check the function subexpression to resolve a type for it if
     // possible.
     fnExpr = typeCheckChildIndependently(callExpr->getFn());
-    if (!fnExpr)
-      return true;
+    if (!fnExpr) {
+      auto hadError = [&]() {
```

Comment:
Could you please  extract this logic into separate static function something like `diagnoseSubscriptMisuse` or similar? `visitApplyExpr` is already pretty big.

Reply:
Indeed. I had a second though about moving everything. Not ideal but hopefully it'll shrink and be part of `diagnoseMemberFailure` later on.

Reply:
Right, that would be best, but we got to work with what we have :)